### ISSUE TITLE
getStreamAsArrayBuffer from a ReadableStream fails on Chrome, but works on FF.

### ISSUE THREAD

**Issue by emctague** [sindresorhus/get-stream#116]
Given the following snippet:

```ts
const inputString = "any text here"
const output = await getStreamAsArrayBuffer(
      new Blob([inputString]).stream().pipeThrough(new CompressionStream('gzip')))
```

This will succeed on Firefox, but fail on Chrome, apparently because `isAsyncIterable` returns false when given the resultant `ReadableStream` in that browser. It appears that Chrome's ReadableStream is [not yet async iterable](https://issues.chromium.org/issues/40612900).

The README for the repository seems to indicate that a ReadableStream is valid input for `getStreamAsArrayBuffer`, and that it should work on all modern browsers which support ReadableStream.

**Comment by ehmicky** [comment]
Thanks @emctague, that's a great (yet unfortunate) finding.

The logic in this module completely relies on async iterations. Actually, while it is called `get-stream`, it does not do anything stream-specific, instead it behaves as `get-async-iterable`. This module started as a Node.js-specific package targeting only Node.js streams, and generalized it to any async iterables in order to bring web support, since both web streams and Node.js streams are async iterables.

Polyfilling async iteration for web streams is [quite complicated](https://github.com/MattiasBuelens/web-streams-polyfill/blob/master/src/lib/readable-stream/async-iterator.ts). With Node.js streams, it's solvable in [~20 lines of code](https://github.com/nodejs/node/blob/a51efa2bcfb415539c3d301b90f932858d36928f/lib/internal/streams/readable.js#L1357) but support for web streams would be quite tedious.

It seems like Chrome has started working on this [since last week](https://issues.chromium.org/issues/40612900#comment19). Unfortunately, support might still be lagging for some time [on Safari and Opera](https://developer.mozilla.org/en-US/docs/Web/API/ReadableStream#browser_compatibility). Node.js, Deno and Firefox do currently support it.

This does not impact users who only target Node.js/Deno, or users who target web browsers but only use async iterables (instead of web streams). 

A polyfill and a ponyfill [are available](https://github.com/MattiasBuelens/web-streams-polyfill). I would suggest the following: we should add a browser-specific entrypoint that uses that ponyfill. The ponyfill is pretty big ([67KB minified](https://bundlephobia.com/package/web-streams-polyfill@3.3.3)) but web browser support would otherwise be basically broken if this works only on Firefox. A big browser entrypoint would be an improvement over a small entrypoint that does not work.

An alternative would be to ask users to install that polyfill/ponyfill themselves. However, if almost every web browser user needs to do it, I guess we might as well do it on their behalf?

What are your thoughts on this @sindresorhus?

**Comment by ehmicky** [comment]
It appears that `ReadableStream[Symbol.asyncIterator]` has been implemented and [will be released with Chrome 124](https://issues.chromium.org/issues/40612900#comment21). However, it's still missing with Edge, Safari and Opera.

Implementing it ourselves is possible, but is definitely quite complicated, especially if we want to thoroughly test every edge case.

Adding the ponyfill mentioned above is also a problem, even if we used a different entrypoint because [it is 431KB](https://packagephobia.com/result?p=web-streams-polyfill) which would be noticeable during `npm install`.

The only other ponyfill I found is the [following one](https://github.com/Sec-ant/readable-stream), which I will look into to make sure it covers all edge cases. I will report back once done.
